packages/hash/api/src/db/adapter.ts

```diff
@@ -541,39 +542,80 @@ export interface DbClient {
     emailAddress: string;
   }): Promise<VerificationCode>;
 
+  /**
+   * Create an aggregation for an entity.
+   *
+   * @param params.sourceAccountId the account id of the source entity
+   * @param params.sourceEntityId the entity id of the source entity
+   * @param params.path the aggregation path
+   * @param params.operation the aggregation operation
+   * @param params.createdByAccountId the account id of the user that created the aggregation
+   */
   createAggregation(params: {
-    createdByAccountId: string;
     sourceAccountId: string;
     sourceEntityId: string;
     path: string;
     operation: object;
+    createdByAccountId: string;
   }): Promise<DbAggregation>;
 
+  /**
+   * Update the operation of an existing aggregation.
+   *
+   * @param params.aggregationId the id of the aggregation
+   * @param params.operation the updated aggregation operation
+   */
   updateAggregationOperation(params: {
-    sourceAccountId: string;
-    sourceEntityId: string;
-    path: string;
+    aggregationId: string;
     operation: object;
   }): Promise<DbAggregation>;
 
-  getEntityAggregation(params: {
+  /**
+   * Get an aggregation by its id.
+   *
+   * @param params.aggregationId the id of the aggregation
+   */
+  getAggregation(params: {
+    aggregationId: string;
+  }): Promise<DbAggregation | null>;
+
+  /**
+   * Get an aggregation by its source entity and path.
+   *
+   * @param params.sourceAccountId the account id of the source entity
+   * @param params.sourceEntityId the entity id of the source entity
+   * @param params.sourceEntityVersionId the entityVersionId of the source entity (optional)
+   * @param params.path the aggregation path
+   */
+  getEntityAggregationByPath(params: {
     sourceAccountId: string;
     sourceEntityId: string;
     sourceEntityVersionId?: string;
     path: string;
   }): Promise<DbAggregation | null>;
 
+  /**
+   * Get all aggregations for an entity.
+   *
+   * @param params.sourceAccountId the account id of the source entity
+   * @param params.sourceEntityId the entity id of the source entity
+   * @param params.sourceEntityVersionId the entityVersionId of the source entity (optional)
```

Comment:
I don't think we do this anywhere, but there's a JSDOc [syntax for optional parameters](https://jsdoc.app/tags-param.html#optional-parameters-and-default-values)

Reply:
Interesting, thanks for sharing. I don't want to spend too much time on this documentation for now given it's going to be changed further in the follow-up PRs mentioned in this PR's description, but will ensure that the updated documentation is more JSDoc compliant.